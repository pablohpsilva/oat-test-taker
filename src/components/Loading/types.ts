export default interface ILoading {
    size?: number
}