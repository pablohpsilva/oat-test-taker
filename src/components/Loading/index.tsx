import React from 'react'
import styled from 'styled-components/macro'

import loadingSVG from '@/assets/animation/loading.svg'

const StyledWrapper = styled.div`
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    padding: var(--dim-gutter);
`

const Loading = ({ size = 30 }) => (
    <StyledWrapper>
        <picture>
            <img src={loadingSVG} alt="loading..." height={size} />
        </picture>
    </StyledWrapper>
)

export default Loading
