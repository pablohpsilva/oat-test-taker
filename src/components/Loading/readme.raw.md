# Loading

## Props

| Property | Description  | Type     | Default |
| -------- | ------------ | -------- | ------- |
| `size`   | Loading size | `number` | `30`    |
