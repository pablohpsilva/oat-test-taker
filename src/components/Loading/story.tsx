import React from 'react'
import { storiesOf } from '@storybook/react'
import { text, number } from '@storybook/addon-knobs'

import { Loading } from '@/components'
import readme from './readme.raw.md'

const LoadingComp = ({ size = number('size', 30) }) => <Loading size={size} />

const opts = {
    notes: { markdown: readme }
}

storiesOf('Components|Loading', module)
    .addDecorator(storyFn => <div>{storyFn()}</div>)
    .add('Example', () => <LoadingComp />, opts)
