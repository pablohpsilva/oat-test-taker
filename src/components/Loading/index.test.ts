import React from 'react'
import Loading from './'

test('should render the Loading component', async () => {
    const wrapper = shallow(<Loading />)

    expect(wrapper.first().hasClass('Loading')).toMatchSnapshot()
    expect(wrapper).toMatchSnapshot()
})
