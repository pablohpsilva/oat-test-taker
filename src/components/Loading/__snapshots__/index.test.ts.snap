// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Loading component 1`] = `false`;

exports[`should render the Loading component 2`] = `
<Loading__StyledWrapper>
  <Icon
    alt="loading..."
    size={30}
    src={null}
  />
</Loading__StyledWrapper>
`;
